runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/sideinput/SideInputStateFetcher.java

```diff
@@ -0,0 +1,257 @@
+    @SuppressWarnings({
+      "deprecation" // Underlying ViewFn is needed to validate the materialization URN.
+    })
+    String materializationUrn = view.getViewFn().getMaterialization().getUrn();
+
+    checkState(
+        SUPPORTED_MATERIALIZATIONS.contains(materializationUrn),
+        "Only materialization's of type %s supported, received %s",
+        SUPPORTED_MATERIALIZATIONS,
+        materializationUrn);
+  }
+
+  private <T> SideInput<T> createSideInputCacheEntry(PCollectionView<T> view, GlobalData data)
+      throws IOException {
+    @SuppressWarnings({
+      "deprecation" //  Materialization URN is required to create the SideInputCacheEntry
+    })
+    String viewMaterializationUrn = view.getViewFn().getMaterialization().getUrn();
+    Iterable<?> rawData = decodeRawData(view, data);
+
+    switch (viewMaterializationUrn) {
+      case ITERABLE_MATERIALIZATION_URN:
+        {
+          @SuppressWarnings({
+            "unchecked", // ITERABLE_MATERIALIZATION_URN has ViewFn<IterableView, T>.
+            "deprecation", // Underlying ViewFn is required to convert the rawData
+            "rawtypes" //  TODO(https://github.com/apache/beam/issues/20447)
+          })
+          ViewFn<IterableView, T> viewFn = (ViewFn<IterableView, T>) view.getViewFn();
+          return SideInput.ready(viewFn.apply(() -> rawData), data.getData().size());
+        }
+      case MULTIMAP_MATERIALIZATION_URN:
+        {
+          @SuppressWarnings({
+            "unchecked", // MULTIMAP_MATERIALIZATION_URN has ViewFn<MultimapView, T>.
+            "deprecation", // Underlying ViewFn is required to convert the rawData
+            "rawtypes" //  TODO(https://github.com/apache/beam/issues/20447)
+          })
+          ViewFn<MultimapView, T> viewFn = (ViewFn<MultimapView, T>) view.getViewFn();
+
+          @SuppressWarnings({
+            "deprecation" // internal coder's key coder is required to convert the rawData into T
+          })
+          Coder<?> keyCoder = ((KvCoder<?, ?>) view.getCoderInternal()).getKeyCoder();
+
+          @SuppressWarnings({
+            "unchecked", // Safe since multimap rawData is of type Iterable<KV<K, V>>
+            "rawtypes", // TODO(https://github.com/apache/beam/issues/20447)
+          })
+          T multimapSideInputValue =
+              viewFn.apply(
+                  InMemoryMultimapSideInputView.fromIterable(keyCoder, (Iterable) rawData));
+          return SideInput.ready(multimapSideInputValue, data.getData().size());
+        }
+      default:
+        {
+          @SuppressWarnings({
+            "deprecation" // ViewFn is required to get the underlying materialization URN.
+          })
+          String unknownMaterializationFormatErrorMessage =
+              String.format(
+                  "Unknown side input materialization format requested '%s'",
+                  view.getViewFn().getMaterialization().getUrn());
```

Comment:
just use viewMaterializationUrn and remove suppression here.

Reply:
done.